Include/pyerrors.h

```diff
@@ -224,6 +224,9 @@ PyAPI_FUNC(void) PyErr_WriteUnraisable(PyObject *);
 /* In signalmodule.c */
 PyAPI_FUNC(int) PyErr_CheckSignals(void);
 PyAPI_FUNC(void) PyErr_SetInterrupt(void);
+#if !defined(Py_LIMITED_API) || Py_LIMITED_API+0 >= 0x030A0000
+PyAPI_FUNC(int) PyErr_SetInterruptEx(int signum);
```

Comment:
I would prefer to exclude it from the limited C API: define it in Include/cpython/pyerrors.h.

About the name, I dislike "Ex" suffix. What about "PyErr_RaiseSignal(signum)"?

Reply:
I'd rather keep the name similar, since those functions are closely related. Also, `PyErr_RaiseSignal` sends the wrong message (since the signal is _not_ raised).

Reply:
Hmm, as for excluding it from the limited C API, it doesn't seem to make sense: `PyErr_SetInterrupt` is in it. And I don't think it places a burden.

Reply:
If you want to add a function to the stable ABI, you must run "make regen-limited-abi" and add it to PC/python3dll.c.

> And I don't think it places a burden.

Other Python implementations must implement it, it's a maintenance burden for them. `_thread.interrupt_main(signum)` can be called in C.

If you don't need the function, I would even suggest to remove it (make it internal/private).

Reply:
You can't call `_thread.interrupt_main` from a signal handler.

Reply:
> Other Python implementations must implement it, it's a maintenance burden for them

It's almost zero-cost if `PyErr_SetInterrupt` is already implemented. `PyErr_SetInterrupt` is the potentially bothersome part, and it's already there.